File: Java/src/Evaluate Reverse Polish Notation.java
public class Solution {
    public int evalRPN(String[] tokens) {
        Stack<String> stack = new Stack<String>();
        for(String s:tokens){
            if(s.equals("+") || s.equals("-") || s.equals("*") || s.equals("/")){
                int ret = 0;
                int last = Integer.parseInt(stack.pop());
                int sec = Integer.parseInt(stack.pop());
                if (s.equals("*")){
                    ret = sec * last;
                }
                if (s.equals("/")){
                    ret = sec / last;
                }
                if (s.equals("+")){
                    ret = sec + last;
                }
                if (s.equals("-")){
                    ret = sec - last;
                }
                stack.push(Integer.toString(ret));
            }
            else{
                stack.push(s);
            }
        }
        
        return Integer.parseInt(stack.pop());
    }
}

File: Python/Evaluate Reverse Polish Notation.py
class Solution:
    # @param tokens, a list of string
    # @return an integer
    def evalRPN(self, tokens):
        value = []
        op = ("+","-","*","/")
        for i in tokens:
            if i in op:
                last = value.pop()
                sec = value.pop()
                if i == "+":
                    value.append(sec + last)
                elif i == "*":
                    value.append(sec * last)
                elif i == "-":
                    value.append(sec - last)
                elif i == "/":
                    value.append(int(float(sec) / last))
            else:
                value.append(int(i))
        return value[-1]